src/Illuminate/Database/Eloquent/Collection.php

```diff
@@ -36,6 +36,47 @@ public function find($key, $default = null)
     }
 
     /**
+     * Determine if the given relation is loaded on collection models.
+     *
+     * @param  string  $key
+     * @return bool
+     */
+    public function relationLoaded($key)
+    {
+        // If collection is empty relations are still loaded, but nothing was found
+        if ($this->isEmpty()) {
+            return true;
+        }
```

Comment:
If you do
```php
$collection->load('foo.bar');
```
but `bar` does not exist on any `foo`-collection, the relation is still loaded, but nothing was found. But yes, if `$collection` is empty I agree that `true`should not be returned.